ecdsa/src/signing.rs

```diff
@@ -539,3 +542,13 @@ where
         Self::from_pkcs8_pem(s).map_err(|_| Error::new())
     }
 }
+
+#[cfg(feature = "algorithm-identifier")]
+impl<C> AssociatedAlgorithmIdentifier for SigningKey<C>
+where
+    C: PrimeCurve + AssociatedAlgorithmIdentifier + CurveArithmetic,
+    Scalar<C>: SignPrimitive<C>,
+    SignatureSize<C>: ArrayLength<u8>,
+{
+    const ALGORITHM_IDENTIFIER: AlgorithmIdentifierRef<'static> = C::ALGORITHM_IDENTIFIER;
```

Comment:
I think instead of this we can add a generic impl of `AssociatedAlgorithmIdentifier` to `SecretKey<C>` and `PublicKey<C>` in the `elliptic-curve` crate, then propagate that to `SigningKey<C>` and `VerifyingKey<C>`.